src/test/ui/issues/issue-105138.rs

```diff
@@ -0,0 +1,20 @@
+// Regression test for #105138.
```

Comment:
Please move this to a better directory in `ui/` and give it a more descriptive name. Afterwards, this can be approved.

Reply:
Moved to `src/test/ui/trait-bounds/impl-bound-with-references-error.rs` 👍